### ISSUE TITLE
Investigate optimistically moving items

### ISSUE THREAD

**Issue by bhollis** [DestinyItemManager/DIM#2078]
We may be able to optimize for the perception of speed even when we're constrained by throttling or network latency, by moving items in the UI before the request to move them completes.

**Comment by bhollis** [comment]
I'm skeptical of this approach for three reasons:

1. Bungie.net can move an item in about 75ms round trip. That's effectively instant to a user.
2. Our move logic is complex - applying a loadout may require many, many moves as we make room for items across three characters and the vault. At any point this may go wrong. Keeping track of the queue of moves and synchronizing an optimistic view state based on that will be fraught with peril.
3. If we *do* have to reverse a move, that's likely to be very confusing to a user. I also personally like that, while I'm playing the game, items drop into the in-game UI at the same time they move in DIM. If I applied a loadout and it said "all done" and then I saw items filtering in over a few seconds on my screen, I think I'd be annoyed.

Most of our latency comes from self-throttling. If we can redesign throttling to take advantage of Bungie.net's actual throttling algorithm, we can get closer to the theoretical maximum move speed. Then, this may be a non-issue.
